workflows/ncn/worker/worker.rebuild.yaml

```diff
@@ -0,0 +1,246 @@
+                    # set last rebuilding worker to nodeMoveTo
+                    nodeMoveTo=${targetNcnsArray[-1]}
+                    # try to find a node that is not in targetNcns
+                    nodes=$(kubectl get nodes | grep "ncn-w" | awk '{print $1}')
+                    for node in $nodes;do
+                      if echo $targetNcns | grep $node; then
+                        echo "skip $node"
+                      else
+                        nodeMoveTo=${node}
+                        break
+                      fi
+                    done
+
+                    echo
+
+                    pods=( "nexus" "ipxe" "kea" "cray-cfs-api-db" )
+                    for pod in ${pods[@]};do
+                      podName=$(kubectl get pods --all-namespaces | awk '{print $2}' | grep $pod)
+                      if [[ -z $podName ]];then
+                        echo "Couldn't not find pod: $pod"
+                        echo "something is wrong"
+                        exit 1
+                      fi                 
+                      
+                      currentNode=$(kubectl get po -A -o wide | grep $podName | awk '{print $8}')
+                      if [[ "$nodeMoveTo" != "$currentNode" ]]; then
+                        echo "Move Pod: $podName to Node: $nodeMoveTo"
+                        /opt/cray/platform-utils/move_pod.sh $podName $nodeMoveTo > /dev/null
+                      fi
+
+                      podName=$(kubectl get pods --all-namespaces | awk '{print $2}' | grep $pod)
+                      ns=$(kubectl get po -A | grep $podName | awk '{print $1}')
+                      kubectl wait --for=condition=ready pod $podName -n $ns --timeout=5m
+                    done
+          {{- range $index,$value := .TargetNcns }}
+          - name: add-labels-{{$value}}
+            template: add-labels
+            arguments: 
+              parameters:
+              - name: targetNcn
+                value: {{$value}}
+          # drain: sync
+          #     Only one worker can be drained at a time
+          - name: drain-{{$value}}
+            template: drain
+            dependencies:
+              - add-labels-{{$value}}
+              - move-critical-singleton-pods
+              - install-csi
+              # each drain depends on previous drain action
+              # so we make sure only one node is drained at a time
+              {{ if ne $index 0 }}
+              - drain-{{ index $.TargetNcns (add $index -1) }}
+              {{ end }}
+            arguments:
+              parameters:
+              - name: targetNcn
+                value: {{$value}}
+              - name: dryRun
+                value: "{{$.DryRun}}"
+          # wipe and reboot: parallel
+          #     once a worker node is drained from k8s
+          #     we can safely wipe and reboot this node
+          #     regardless of what state other nodes are
+          - name: wipe-and-reboot-{{$value}}
```

Comment:
We don't have to wipe anymore, we can though.

This should probably just be "set nextboot"

It should just set pxe for the next boot, and maybe disable the disk bootloader.

For reference the rebuild script in main is updated to do just this, I'd link it for reference right now if I could but I'm mobile. I'll link it Monday unless someone beats me to it.

Reply:
Wiping would ensure a pxe boot, but the reason I care is I want to remove the mentality that a wipe is required. The deployment code will wipe. If a wipe fails then it's the fault of the deployment code, and the deployment code handles wiping better than the manual/argo step(s).

Reply:
i still have that ticket assigned to me. the reason why i kept this for now is so i can test on a 1.2 system until we get all 1.3 upgrade script cleaned up